tools/scripts/uconfig_vars_test.py

```diff
@@ -0,0 +1,217 @@
+
+    Args:
+      uconfig_file: common/unicode/uconfig.h as string.
+
+    Returns:
+      List of all UCONFIG_NO_XXX flags found in uconfig.h, initialized test
+      result structure.
+    """
+
+    uconfig_no_flags_all = []
+    test_results = {}
+    uconfig_no_regex = r'UCONFIG_NO_[A-Z_]*'
+
+    for uconfig_no_flag in re.finditer(uconfig_no_regex, uconfig_file):
+        if uconfig_no_flag.group(0) not in uconfig_no_flags_all:
+            uconfig_no_flags_all.append(uconfig_no_flag.group(0))
+
+    test_results = {f: {'unit_test': False, 'hdr_test': False} for f in uconfig_no_flags_all}
+    test_results['all_flags'] = {'unit_test': False, 'hdr_test' : False}
+
+    return uconfig_no_flags_all, test_results
+
+def EnableUConfigNo(uconfig_file, uconfig_no_flag):
+    """Sets the uconfig_no_flag in common/unicode/uconfig.h file.
+
+    Args:
+      uconfig_file: common/unicode/uconfig.h as string.
+      uconfig_no_flag: flags that is to be set to 1.
+
+    Returns:
+      uconfig.h with flag set to 1
+    """
+
+    uconfig_no_def_regex = r'(?m)#ifndef %s\n#\s+define %s\s+0\n#endif$' % (
+        uconfig_no_flag, uconfig_no_flag)
+    uconfig_no_def_match = re.search(uconfig_no_def_regex, uconfig_file)
+    if not uconfig_no_def_match:
+        print('No definition for flag %s found!\n' % uconfig_no_flag)
+        sys.exit(-1)
+    uconfig_no_def = uconfig_no_def_match.group(0)
+    set_uconfig_no_def = uconfig_no_def.replace(' 0\n', ' 1\n')
+    set_in_uconfig_file = uconfig_file.replace(uconfig_no_def, set_uconfig_no_def)
+
+    return set_in_uconfig_file
+
+def main():
+    # Read the options and determine what to run.
+    run_hdr = False
+    run_unit = False
+    optlist, _ = getopt.getopt(sys.argv[1:], "pu")
+    for o, _ in optlist:
+        if o == "-p":
+            run_hdr = True
+        elif o == "-u":
+            run_unit = True
+
+    os.chdir('icu4c/source')
+    orig_uconfig_file = ReadFile('common/unicode/uconfig.h')
+
+    all_uconfig_no_flags, test_results = ExtractUConfigNoXXX(orig_uconfig_file)
+    if not all_uconfig_no_flags:
+        print('No UCONFIG_NO_XXX flags found!\n')
+        sys.exit(-1)
+    SetUpICU()
+
+    # Iterate over all flags, set each individually one in uconfig.h and
+    # execute ICU4C unit tests or/and header test.
+    for uconfig_no in all_uconfig_no_flags:
+        with open('common/unicode/uconfig.h', 'w') as file_handle:
+            file_handle.write(EnableUConfigNo(orig_uconfig_file, uconfig_no))
+        RunCmd('make clean')
+
+        # Run ICU4C unit tests if requested, except for the excluded flags.
+        if run_unit and uconfig_no not in excluded_unit_test_flags:
+            print('Running unit tests with %s set to 1.' % uconfig_no)
+            _, exit_code = RunCmd('make -j2 check')
+            if exit_code == 0:
+                test_results[uconfig_no]['unit_test'] = True
+            else:
+                test_results[uconfig_no]['unit_test'] = False
```

Comment:
You should be able to replace this with
```
test_results[uconfig_no]['unit_test'] = (exit_code == 0)
```
with or without the parentheses to your liking

Reply:
Done throughout.